src/Flexdis86/Assembler.hs

```diff
@@ -253,29 +255,55 @@ matchOperandType oso ops =
     (QWordReg (Reg64 rno), OpType (Opcode_reg rcode) VSize) -> rno `mod` 8 == rcode
     _ -> False
 
+
+data AssembledInstruction a =
+  AssembledInstruction
+    { legacyPrefixes :: a
```

Comment:
@travitch said:

> I also forget how it is specified, but in the spirit of breaking things up a bit more, I think it would make sense to separate REX from the legacy prefixes (REX/VEX are the non-legacy prefixes).